rust/sbd-server/src/ws/ws_fastwebsockets.rs

```diff
@@ -0,0 +1,91 @@
+use super::Payload;
+use crate::*;
+use std::io::{Error, Result};
+use std::sync::Arc;
+
+/// WebSocket abstraction.
+pub struct WebSocket<S>(
+    fastwebsockets::WebSocket<
+        hyper_util::rt::TokioIo<hyper::upgrade::Upgraded>,
+    >,
+    std::marker::PhantomData<S>,
+)
+where
+    S: tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send + 'static;
+
+impl<S> WebSocket<S>
+where
+    S: tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send + 'static,
+{
+    /// As a server, accept/upgrade a new client connection.
+    pub async fn upgrade(config: Arc<Config>, s: S) -> Result<Self> {
+        let io = hyper_util::rt::TokioIo::new(s);
+        let (s, mut r) = tokio::sync::mpsc::unbounded_channel();
+        let _res = hyper::server::conn::http1::Builder::new()
+            .serve_connection(
+                io,
+                hyper::service::service_fn(move |mut req| {
+                    // TODO report the PUBKEY here
```

Comment:
the fastwebsockets backend currently doesn't even compile, since it was a dead end. I didn't want to loose the work incase it's viable some day, but I don't think it needs to be reviewed right now.

Reply:
i suspect we'll lose that context over time and it deserves to be mentioned in the docstring